design/jira-traceability.md

```diff
@@ -0,0 +1,122 @@
+
+#### Jira Build UI and APIs:
+The  build results are populated in the development section of the work item UI. On further clicking on it, we see the pipeline runs that were associated with the build. 
+
+![Build1](./images/jira-traceability-build-1.png)
+
+
+Drill down view:
+![Build2](./images/jira-traceability-build-2.png)
+
+For the API, the combination of `pipelineId` (pipeline definition identifier) and `buildNumber` (pipeline run identifier) is considered as the unique identifier for a build. 
+
+#### Jira Deployment UI and APIs:
+
+Jira has a separate section for Deployments in its work item UI. In this, only the highest deployed environment category is displayed. Eg. If the work item was deployed to a `test` type environment and  then to `production` type, Jira shows only `production` in deployment section. On clicking on this entry the user can see details of previous deployments.
+
+![Release1](./images/jira-traceability-deploy-1.png)
+
+Drill down view:
+![Release2](./images/jira-traceability-deploy-2.png)
+
+For the API, the combination of `pipelineId` (pipeline definition identifier), `environmentID` (environment identifier), `deploymentSequenceNumber` (deployment identifier) constitute a unique identifier for deployment. 
+
+
+#### When should the data be pushed from Azure Pipelines:?
+ Since build generation and deployments are done at a job level in azure pipelines, we'll push the status information to Jira at job boundaries. For deployments, azure pipelines already has a separate category of jobs, viz, deployment jobs. We'll invoke deployment APIs at the beginning and end of deployment jobs if all the filter conditions are met. For builds, however, we currently do not have a separate category of jobs. For this, we will assume that any job that publishes an artifact is a build job. We'll invoke build API at the end of these jobs if filter conditions are met.  
+
+
+
+## YAML: 
+
+Since traceability primarily involves interacting with external systems, we propose taking configuration for interaction with these external systems in a separate `integrations` section in YAML. As a beginning, we will support a first class integration with JIRA. In future, we may eventually make this section extensible thereby allowing 3rd party authors to create their own integrations. In this document, we will keep focus on JIRA integration. <em>Details on extensibility design are out of scope for this document.</em>
+
+ 
+
+### Schema:
+
+```yaml
+integrations:       
```

Comment:
This should look more like:

```yaml
integrations:
- integration: 'jira'
  connection: string
  #...
```

Reply:
(As you've currently written it, there can be only one integration since it's a property of the "integrations" object. I think you meant "this is a list of integrations".)

Reply:
I have updated the schema to reflect this.